frontend/src/views/TimetableView.vue

```diff
@@ -45,20 +58,37 @@ function handleDownloadAll() {
     <div v-else>
       <header class="rsprd-bar">
         <div>
-          <a class="rsprd-link">By class</a>
-          <a class="rsprd-link">By teacher</a>
-          <a class="rsprd-link">By classroom</a>
+          <a
+            @click="filter = timetableFilters.BY_CLASS"
+            :class="{ 'rsprd-link--active': filter === timetableFilters.BY_CLASS }"
+            class="rsprd-link"
+            >By Class</a
+          >
+          <a
+            @click="filter = timetableFilters.BY_TEACHER"
+            :class="{ 'rsprd-link--active': filter === timetableFilters.BY_TEACHER }"
+            class="rsprd-link"
+            >By Teacher</a
+          >
+          <a
+            @click="filter = timetableFilters.BY_CLASSROOM"
+            :class="{ 'rsprd-link--active': filter === timetableFilters.BY_CLASSROOM }"
+            class="rsprd-link"
+            >By Classroom</a
+          >
```

Comment:
Nitpick 1: the link could be a component. It is starting to have a common interface.
Nitpick 2: there could have been an array of filters and they could have been looped here instead of repeating the link three times.
```vue
<rsprd-link v-for="option in filterOptions" :active="filter === option.name">
  {{ option.label }}
</rsprd-link>
```